spec/connect/client_spec.rb

```diff
@@ -5,7 +5,16 @@
   let(:default_logger) { SUSE::Connect::GlobalLogger.instance.log }
   let(:string_logger) { ::Logger.new(StringIO.new) }
   let(:client_instance) { described_class.new config }
-  let(:product) { SUSE::Connect::Remote::Product.new identifier: 'SLES', version: '12', arch: 'x86_64' }
+
+  let(:product_tree) { JSON.parse(File.read('spec/fixtures/product_tree.json')) }
+  let(:product) { Remote::Product.new(product_tree) }
+
+  let(:extension) { product.extensions[1] }
+  let(:subextension) { extension.extensions[1] }
+
+  let(:recommended3) { product.extensions[2] }
+  let(:extension3) { product.extensions[3] }
+  let(:extension5) { extension3.extensions[1] }
```

Comment:
Naming of all these could be better. You have `extension` and `subextensions` without numbers, then you have `recommended3` and `extension3`. And none of these correspond with array indices in the square brackets.

(See the comment below about identifiers in the fixture -- name things here and in the fixture the same, it will simplify things greatly)